### ISSUE TITLE
Healthcheck filter fails when disabling server metrics

### ISSUE THREAD

**Issue by ivanpondal** [envoyproxy/envoy#8630]
Given an **Envoy v1.11.1** node with [server](https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/statistics#server) metrics excluded via `StatsMatcher` and `HTTP health checking filter` enabled, when hitting the healthcheck endpoint a 503 response is received independently of the true node health.

From my research this seems to be due to the healthcheck depending directly on the server metrics:
- [Health check filter](https://github.com/envoyproxy/envoy/blob/v1.11.1/source/extensions/filters/http/health_check/health_check.cc#L120)
- [healthCheckFailed() instance method](https://github.com/envoyproxy/envoy/blob/v1.11.1/source/server/server.cc#L207)

### Minimal config to reproduce issue
```
admin:
  access_log_path: /tmp/admin_access.log
  address:
    socket_address: { address: 0.0.0.0, port_value: 9901 }
stats_config:
  stats_matcher:
    reject_all: true
static_resources:
  listeners:
  - name: proxy
    address:
      socket_address: { address: 0.0.0.0, port_value: 10000 }
    filter_chains:
    - filters:
      - name: envoy.http_connection_manager
        config:
          stat_prefix: ingress_http
          route_config:
            name: route
          http_filters:
          - name: envoy.health_check
            config:
              pass_through_mode: false
              headers:
              - name: ":path"
                exact_match: "/healthcheck"
          - name: envoy.router
```
### Healthcheck request
```
$ curl -i localhost:10000/healthcheck
HTTP/1.1 503 Service Unavailable
x-envoy-upstream-healthchecked-cluster: local-cluster
date: Wed, 16 Oct 2019 17:33:43 GMT
server: envoy
connection: close
content-length: 0
```

## Workaround
The workaround is to whitelist metrics needed by the health-check:
```
stats_config:
 stats_matcher:
  inclusion_list:
    patterns:
    - suffix: live
    - suffix: membership_healthy
    - suffix: membership_degraded
    - suffix: membership_total
```

**Comment by mattklein123** [comment]
Dup of https://github.com/envoyproxy/envoy/issues/8473

**Comment by venilnoronha** [comment]
This is fixed in #8482.

**Comment by ivanpondal** [comment]
@venilnoronha I took a look at the PR and noticed it removes the usage of `server_stats.live` but not the cluster ones (`membership_total`, `membership_degraded ` and `membership_healthy`). Couldn't that cause the minimum healthy clusters functionality to work improperly?
